sqlmesh/core/engine_adapter/spark.py

```diff
@@ -237,18 +237,21 @@ def create_view(
     def _create_table_properties(
         self,
         storage_format: t.Optional[str] = None,
-        partitioned_by: t.Optional[t.List[str]] = None,
+        partitioned_by: t.Optional[t.List[exp.Expression]] = None,
         partition_interval_unit: t.Optional[IntervalUnit] = None,
     ) -> t.Optional[exp.Properties]:
         format_property = None
         partition_columns_property = None
         if storage_format:
             format_property = exp.FileFormatProperty(this=exp.Var(this=storage_format))
         if partitioned_by:
+            for expr in partitioned_by:
+                if not isinstance(expr, exp.Column):
+                    raise SQLMeshError(
+                        f"PARTITIONED BY contains non-column value '{expr.sql(dialect='spark')}'."
+                    )
             partition_columns_property = exp.PartitionedByProperty(
-                this=exp.Schema(
-                    expressions=[exp.to_identifier(column) for column in partitioned_by]
-                ),
+                this=exp.Schema(expressions=partitioned_by),
```

Comment:
Is it ok that that this is a list of `Column` vs list of identifiers like it used to be?

Reply:
yes, verified via unit tests.